src/TypeFormatter/LiteralUnionTypeFormatter.ts

```diff
@@ -43,33 +43,41 @@ export class LiteralUnionTypeFormatter implements SubTypeFormatter {
         });
 
         if (allStrings && hasString && !preserveLiterals) {
-            return {
-                type: hasNull ? ["string", "null"] : "string",
-            };
+            return hasNull ? { type: ["string", "null"] } : { type: "string" };
         }
 
-        const values = uniqueArray(types.flatMap(getLiteralValues));
-        const typeNames = uniqueArray(types.flatMap(getLiteralTypes));
+        const typeValues: Set<LiteralValue | null> = new Set();
+        const typeNames: Set<RawTypeName> = new Set();
 
-        const ret = {
-            type: typeNames.length === 1 ? typeNames[0] : typeNames,
-            enum: values,
-        };
+        for (const type of types) {
```

Comment:
I actually liked this being in a separate method. You can make a set from an array easily if you want to use the original method.

Reply:
4365483

Reply:
thanks. feel free to merge when you think it's ready